### PR TITLE
Mark PWM and Servo Pins

### PR DESCRIPTION
**Pull request by brentru**
Uses `hasPWM` boolean field to address if a pin within an array supports PWM, or is **Analog-Input Only** (or an input-only sensor is connected to the pin, such as a light sensor).

The `pwm` Wippersnapper_Components on the Adafruit IO interface must only display the `hasPWM: true` pins.

Examples added for ESP32, ESP8266 and SAMD51 platforms.

Resolves https://github.com/adafruit/Wippersnapper_Boards/issues/68

cc @ladyada  - does this make sense?

### CODE REVIEW

**Comment by brentru** [comment]
After the PR has been reviewed, I'll add this to all boards within this repo.

**Comment by brentru** [comment]
Marking this property `false` by default

**Comment by brentru** [comment]
note: possibly switch to `hasPWM` to remove ambiguity around analog pins without and with PWM support.

**Comment by brentru** [comment]
@lorennorman  Re-requesting a review. I've covered the Adafruit boards here and added both pwm and servo flags.

We'll want to make an issue for adding these flags to the non-adafruit boards (possibly a good hacktoberfest issue).

**Comment by ladyada** [comment]
@brentru woops i didnt see the tag above, do you still need me to 👁️

**Comment by brentru** [comment]
@ladyada  nope! worked it out with Loren

File: boards/pyportal-tinyusb/definition.json [hunk 1]
```diff
@@ -24,13 +24,31 @@
...
+        {
+            "name": "A0",
+            "displayName": "A0 (Speaker)",
+            "dataType": "int16",
+            "hasPWM": true
+        },
         {
             "name": "A1",
-            "displayName": "A1",
+            "displayName": "D3",
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/pyportal-tinyusb/definition.json:51]: It is, was wrong initially.

File: boards/pyportal-tinyusb/definition.json [hunk 2]
```diff
@@ -39,8 +57,8 @@
             "dataType": "int16"
         },
         {
-            "name": "A4",
-            "displayName": "A4",
+            "name": "A3",
+            "displayName": "D4",
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/pyportal-tinyusb/definition.json:61]: Correct, was initially wrong.

File: boards/feather-esp32-v2/definition.json [hunk 3]
```diff
@@ -8,10 +8,10 @@
     "documentationURL":"https://learn.adafruit.com/adafruit-esp32-feather-v2",
     "installMethod":"web",
     "esptool": {
-        "fileSystemSize": 1638400,
-        "blockSize": 4096,
-        "offset": "0x670000",
         "chip": "esp32",
+        "fileSystemSize": 94208,
+        "blockSize": 4096,
+        "offset": "0x290000",
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32-v2/definition.json:14]: hm, these should be set to 
```
        "fileSystemSize": 1638400,
        "blockSize": 4096,
        "offset": "0x670000",
        "chip": "esp32",
```

Fixing...

File: boards/feather-esp32-v2/definition.json [hunk 4]
```diff
@@ -24,67 +24,65 @@
...
-       },
-       {
-        "name":"D36",
-        "displayName":"D36",
-        "dataType":"bool"
-       },
-       {
-        "name":"D4",
-        "displayName":"D4",
-        "dataType":"bool"
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32-v2/definition.json:87]: Unused!

File: boards/feather-esp32-v2/definition.json [hunk 5]
```diff
@@ -127,42 +127,9 @@
...
-       },
-       {
-          "name":"A11",
-          "displayName":"A11",
-          "dataType":"int16"
-       },
-       {
-          "name":"A12",
-          "displayName":"A12",
-          "dataType":"int16"
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32-v2/definition.json:165]: Unused pins + did not match silkscreen

File: boards/feather-esp32/definition.json [hunk 6]
```diff
@@ -22,66 +22,105 @@
...
+         "name":"D36",
+         "displayName":"D34 (A4)",
+         "dataType":"bool"
+       },
+       {
+         "name":"D4",
+         "displayName":"D4 (A5)",
+         "dataType":"bool",
+         "hasPWM":true,
+         "hasServo":true
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32/definition.json:107]: Yeah, new pins matching silk

File: boards/feather-esp32s2-tft/definition.json [hunk 7]
```diff
@@ -9,119 +9,123 @@
...
-          },
-          {
-             "name":"D7",
-             "displayName":"D7",
-             "dataType":"bool"
-          },
-          {
-             "name":"D8",
-             "displayName":"D8",
-             "dataType":"bool"
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32s2-tft/definition.json:55]: Not on silkscreen for ESP32-S2 TFT, removed.

File: boards/feather-esp32s2-tft/definition.json [hunk 8]
```diff
@@ -9,119 +9,123 @@
     "installMethod":"uf2",
     "components":{
        "digitalPins":[
-          {
-             "name":"D0",
-             "displayName":"BOOT",
-             "dataType":"bool"
-          },
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32s2-tft/definition.json:16]: added back

File: boards/feather-esp32s2-tft/definition.json [hunk 9]
```diff
@@ -9,119 +9,123 @@
...
           },
           {
              "name":"D13",
              "displayName":"D13 (LED)",
              "dataType":"bool"
-          },
-          {
-             "name":"D14",
-             "displayName":"D14",
-             "dataType":"bool"
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32s2-tft/definition.json:85]: pin not present on silkscreen or aliased, removed.

File: boards/feather-esp32s2/definition.json [hunk 10]
```diff
@@ -12,84 +12,93 @@
...
-          },
-          {
-             "name":"D7",
-             "displayName":"D7",
-             "dataType":"bool"
-          },
-          {
-             "name":"D8",
-             "displayName":"D8",
-             "dataType":"bool"
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32s2/definition.json:50]: pin not present on silkscreen or aliased, removed.

File: boards/feather-esp32s2/definition.json [hunk 11]
```diff
@@ -12,84 +12,93 @@
...
           },
           {
              "name":"D13",
              "displayName":"D13 (LED)",
              "dataType":"bool"
-          },
-          {
-             "name":"D14",
-             "displayName":"D14",
-             "dataType":"bool"
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32s2/definition.json:80]: pin not present on silkscreen or aliased, removed.

File: boards/feather-esp32s3-4mbflash-2mbpsram/definition.json [hunk 12]
```diff
@@ -8,121 +8,157 @@
     "documentationURL":"https://learn.adafruit.com/adafruit-esp32-s3-feather",
     "installMethod":"uf2",
     "components":{
-      "digitalPins":[
-        {
-          "name":"D0",
-          "displayName":"D0 (BOOT)",
-          "dataType":"bool"
-        },
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/feather-esp32s3-4mbflash-2mbpsram/definition.json:16]: Added back.

File: boards/pyportal-tinyusb/definition.json [hunk 13]
```diff
@@ -24,13 +24,31 @@
...
+            "dataType": "bool"
+          }
       ],
       "analogPins": [
+        {
+            "name": "A0",
+            "displayName": "A0 (Speaker)",
+            "dataType": "int16",
+            "hasPWM": true
+        },
```
[the remark this answers is not in the document]
Comment (brentru) [reply boards/pyportal-tinyusb/definition.json:48]: These are new, unique, pins. Added PWM-capable pins such as backlight dimming pin and piezo speaker pin for lighting and music.

**Pull request merged**